### PR TITLE
add upload

### LINKED ISSUE #3: Web server: store file uploaded from client
The server must store the file binary data and the file type. The client must send the file binary data, and may send metadata such as a filename or a file type.

Methods for storage include
1. the server local filesystem
2. the server database, such as in a MySQL LONGBLOB

If the client does not send metadata, the server must detect the file type. See #1

### PR DESCRIPTION
**Pull request by orogenic**
resolve #3

### CODE REVIEW

**Review by anguyen120 — APPROVED** [review]

File: app.py [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
 def hello_world():
     return 'Hello World!'
 
+@app.route('/uploadfile', methods=['GET', 'POST'])
+def uploadfile():
+    if request.method == 'GET':
+        return current_app.send_static_file('uploadfile.html')
+    elif request.method == 'POST':
+         if 'filename' not in request.files:
+             return 400, 'missing file "filename"'
```
Comment (orogenic) [inline app.py:22]: realized this line's wrong. status code comes after response text.

File: app.py [hunk 2]
```diff
@@ -1,12 +1,28 @@
-from flask import Flask
+# vim: set ts=4 sw=4 expandtab:
```
Comment (orogenic) [inline app.py:1]: i'll set up my vimrc and remove the modeline

File: app.py [hunk 3]
```diff
@@ -1,12 +1,28 @@
...
+import binascii
+import os
```
Comment (orogenic) [inline app.py:3]: binascii and os are unused imports: remove

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 /venv/
 .env
-__pycache__/
\ No newline at end of file
+__pycache__/
+/uploads
```
Comment (orogenic) [inline .gitignore:4]: add trailing slash

File: app.py (hunk 2 above)
Comment (orogenic) [inline app.py:1]: configure developer vimrc, remove modeline

File: app.py [hunk 5]
```diff
@@ -1,12 +1,28 @@
-from flask import Flask
+# vim: set ts=4 sw=4 expandtab:
+import binascii
```
Comment (orogenic) [inline app.py:2]: unused, remove

File: app.py (hunk 1 above)
Comment (orogenic) [inline app.py:22]: bug, response text first, status code second

**Pull request merged**